clojure-lsp: 2023.08.06-00.28.06 -> 2024.04.22-11.50.26

Ramblurr opened the pull request:
fixes #308790 

## Description of changes

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ramblurr committed:
clojure-lsp: 2023.08.06-00.28.06 -> 2024.04.22-11.50.26

Release notes: https://github.com/clojure-lsp/clojure-lsp/releases/tag/2024.04.22-11.50.26

Also applied the requested cleanups:

* change sha256 -> hash
* removed withs
* documented where the build args come from
* removed pname usage
* added meta.changelog

getchoo commented:
Result of `nixpkgs-review pr 310027` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clojure-lsp</li>
  </ul>
</details>

pbsds commented on pkgs/development/tools/misc/clojure-lsp/default.nix:
```diff
@@ -12,26 +12,27 @@
   ];
 
-  doCheck = true;
+  doCheck = false;
```
Is this still needed if you bump further? Can you s

Ramblurr replied:
What do you mean by "bump further"?

Ramblurr replied:
Ah.. this is resolved anyways by `--features=clj_easy.graal_build_time.InitClojureClasses`

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1746

Ramblurr commented:
Adding `--features=clj_easy.graal_build_time.InitClojureClasses` fixed the tests, which was a nice find.

superherointj commented on pkgs/development/tools/misc/clojure-lsp/default.nix:
```diff
@@ -12,23 +12,24 @@
     repo = pname;
     rev = version;
-    sha256 = "sha256-wc7M2cPRtdaRzZn3GNu/aCbQ2VqxiDxvu/b7qnBVUBo=";
+    sha256 = "sha256-GyPIFYR+/BZ+vq6+yuer5HoVILXLWNw1sW8XpJ7q4SA=";
```
```suggestion
    hash = "sha256-GyPIFYR+/BZ+vq6+yuer5HoVILXLWNw1sW8XpJ7q4SA=";
```

superherointj commented on pkgs/development/tools/misc/clojure-lsp/default.nix:
```diff
@@ -12,23 +12,24 @@
+  version = "2024.04.22-11.50.26";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
```
```suggestion
    owner = "clojure-lsp";
    repo = "clojure-lsp";
```

superherointj commented:
> meta.changelog = "https://github.com/clojure-lsp/clojure-lsp/releases/tag/${version}";

(This is used by the bot in update script)

superherointj commented:
Consider moving package to `by-name`.

superherointj commented on pkgs/development/tools/misc/clojure-lsp/default.nix:
```diff
@@ -12,23 +12,24 @@
   jar = fetchurl {
     url = "https://github.com/clojure-lsp/clojure-lsp/releases/download/${version}/clojure-lsp-standalone.jar";
-    sha256 = "c301821ac6914999a44f5c1cd371d46b248fe9a2e31d43a666d0bc2656cfdd78";
+    sha256 = "sha256-dB16225A7L3nWplvqlal+5gho+LmqqVGPN9dfasKaPk=";
```
```suggestion
    hash = "sha256-dB16225A7L3nWplvqlal+5gho+LmqqVGPN9dfasKaPk=";
```

Ramblurr committed:
clojure-lsp: move to pkgs/by-name/cl/clojure-lsp and apply nixfmt

Ramblurr commented:
@jlesquembre @superherointj thanks for the feedback.

In the first commit I applied:


    * change sha256 -> hash
    * removed withs
    * documented where the build args come from
    * removed pname usage
    * added meta.changelog

In the 2nd commit I moved the package to `pkgs/by-name` and applied `nixfmt`

jlesquembre commented:
Result of `nixpkgs-review pr 310027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

JohnRTitor commented:
Result of `nixpkgs-review pr 310027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

superherointj commented:
I'm awaiting for the `x86_64-darwin` build.

JohnRTitor reviewed (approved):
LGTM
